### DISCUSSION TITLE
Is it possible to import modules like pathlib?

### DISCUSSION THREAD

**Discussion by OrigamingWasTaken** [pybricks/support#1049] [Q&A]
I want to make a project manager, and I iterate through every file in the "programs" dir (on my computer) so i can import them later on. But everytime i try to import a lib, it says that the lib is not present. How do i import one?

**Comment by dlech** [comment]
Pybricks uses MicroPython, not Python, so in general, only the  libraries available are the ones listed in [the docs](https://docs.pybricks.com/en/stable/). 

If you are using EV3,  the libraries from [micropython-lib](https://github.com/pybricks/micropython-lib) are available, but this contains only a very small subset of the Python standard library.

To iterate directories on EV3 you can use [uos.ilistdir()](https://docs.micropython.org/en/v1.12/library/uos.html#uos.ilistdir).

**Reply by OrigamingWasTaken** [reply]
Mmmmh... i see. I'm using a spike prime tho. What would the iteration be?

**Reply by dlech** [reply]
Pybricks currently doesn't support a file system on SPIKE Prime.

Any modules in the same directory as your main program that are imported by the main program will be included when you download and run the program. So you can do something like:

`main.py`
```python
# other imports
...

import project1
import project2
...

while True:
    # some UI to select a project number
    project_num = ...

    if project_num == 1:
        project1.run()
    elif project_num == 2:
        project2.run()
    ...
```

`project1.py`
```python
# imports
...

def run():
    # main script
    ...
```

**Reply by OrigamingWasTaken** [reply]
Mmmmh... i wanted to recursively import all project in the programs directory so i don't need to modify the code each time. Is there really no way?

**Reply by OrigamingWasTaken** [reply]
*bump*

**Reply by dlech** [reply]
I suppose you could write a Python program that runs on your computer to automatically generate the main program based on the files in the directory.

**Reply by OrigamingWasTaken** [reply]
yeah, that's what i thought. Thanks
